webui/react/src/services/types.ts

```diff
@@ -62,6 +62,23 @@ export interface ExperimentIdParams {
   experimentId: number;
 }
 
+export interface GetExperimentsParams {
+  sortBy?: 'SORT_BY_UNSPECIFIED' | 'SORT_BY_ID' | 'SORT_BY_DESCRIPTION' | 'SORT_BY_START_TIME'
+  | 'SORT_BY_END_TIME' | 'SORT_BY_STATE' | 'SORT_BY_NUM_TRIALS' | 'SORT_BY_PROGRESS'
+  | 'SORT_BY_USER';
+  orderBy?: 'ORDER_BY_UNSPECIFIED' | 'ORDER_BY_ASC' | 'ORDER_BY_DESC';
+  offset?: number;
+  limit?: number;
+  description?: string;
+  labels?: Array<string>;
+  archived?: boolean;
+  states?: Array<'STATE_UNSPECIFIED' | 'STATE_ACTIVE' | 'STATE_PAUSED'
+  | 'STATE_STOPPING_COMPLETED' | 'STATE_STOPPING_CANCELED' | 'STATE_STOPPING_ERROR'
+  | 'STATE_COMPLETED' | 'STATE_CANCELED' | 'STATE_ERROR' | 'STATE_DELETED'>;
+  users?: Array<string>;
+  options?: never;
```

Comment:
**suggestion:** Even though we are not using it now, I think we need `options` to pass other options such as `signal` for an abort controller.
https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort

Reply:
one option is to define another `options` type that follows the native Fetch api:
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch

Here's the relevant available options (it's missing `signal` needed for aborting a request and probably a few other ones, but `signal` would be a good one to have at the minimum)
```
{
    method: 'POST', // *GET, POST, PUT, DELETE, etc.
    mode: 'cors', // no-cors, *cors, same-origin
    cache: 'no-cache', // *default, no-cache, reload, force-cache, only-if-cached
    credentials: 'same-origin', // include, *same-origin, omit
    headers: {
      'Content-Type': 'application/json'
      // 'Content-Type': 'application/x-www-form-urlencoded',
    },
    redirect: 'follow', // manual, *follow, error
    referrerPolicy: 'no-referrer', // no-referrer, *no-referrer-when-downgrade, origin, origin-when-cross-origin, same-origin, strict-origin, strict-origin-when-cross-origin, unsafe-url
    body: JSON.stringify(data) // body data type must match "Content-Type" header
  }
```

Reply:
What you're suggesting/asking here is to:
1. Add a new TS interface for `Options` based on the `JSON` above.
2. Update the `GetExperimentsParams` TS interface to `... options?: Options; ...`
?

Reply:
I take this back, let's defer this change to when we actually need it. :) my bad.